@intern-challenge/one created the deployment detection bot (#1)
Nailujj/Forta_bot_deployment: README.md

> # Bot Function Call Detection

Reviewer: Could be a bit more specific. Something like "Nethermind Creation, Update, Disable Detection Bot", for example.

## Description

> This bot detects specific function calls made by a known `deployer` address. It monitors transactions for

Reviewer: The description could be more specific by mentioning that it will specifically monitor the Nethermind Deployer address and calls to the Forta Agent Registry, and you could also link to the registry contract on Polygonscan.

`Create Agent`, `Update Agent`, and `Disable Agent` function calls.

## Supported Chains

- Polygon

> - Any other EVM-compatible chains

Reviewer: Since the Forta contract live on Polygon, this bot would really only monitor transactions on Polygon.

> This Forta bot monitors specific function calls made by the known Nethermind `Deployer address`. It watches transactions for `Create Agent`, `Update Agent`, and `Disable Agent` function calls directed to the Forta Agent Registry on Polygon.

Reviewer: ```suggestion
This Forta bot monitors specific function calls made by the known Nethermind Deployer address. It watches transactions for `Create Agent`, `Update Agent`, and `Disable Agent` function calls directed to the Forta Agent Registry on Polygon.
```
Not necessary to put "Deployer address" in ticks, as we typically only put variable/function/file names in ticks in the README.

## Alerts

> ### FORTA-CREATE

Reviewer: Typically, the names of the alerts reference the protocol being monitored. However, for Challenge 01, it could be `NETHERMIND-1`, `NETHERMIND-2`, etc.

> - **Description**: Fired when a transaction contains a Create Agent function call by the deployer address.

Reviewer: ```suggestion
- **Description**: Fired when a transaction contains a `createAgent` function call by the deployer address.
```
Not necessary, but suggesting as an additional detail.

> - **Description**: Fired when a transaction contains a Create Agent function call by the deployer address.

Reviewer: ```suggestion
- **Description**: Fired when a transaction contains a `createAgent` function call by the deployer address.
```
Suggestion to put the functions called in ticks in the alert description. Also applicable to the other two alerts.

- **Severity**: Low.
- **Type**: "info".
- **Metadata**:
  - `txHash`: The hash of the transaction.
  - `from`: The address that sent the transaction.
  - `to`: The address that received the transaction.
  - `data`: The input data of the transaction.

### FORTA-UPDATE
- **Description**: Fired when a transaction contains an Update Agent function call by the deployer address.
- **Severity**: Low.
- **Type**: "info".
- **Metadata**:
  - `txHash`: The hash of the transaction.
  - `from`: The address that sent the transaction.
  - `to`: The address that received the transaction.
  - `data`: The input data of the transaction.

> ### FORTA-DISABLE

Reviewer: The challenge didn't explicitly request monitoring for "disable", but we can consider this a bonus.

Author: Thanks, I am aware it wasnt technically a task but wanted to do this too since i was interested how it would work with more different transactions to monitor :)

- **Description**: Fired when a transaction contains a Disable Agent function call by the deployer address.
- **Severity**: Low.
- **Type**: "info".
- **Metadata**:
  - `txHash`: The hash of the transaction.
  - `from`: The address that sent the transaction.
  - `to`: The address that received the transaction.
  - `data`: The input data of the transaction.

## Test Data

The bot behaviour can be verified with the following transactions:

> - Create Agent function call example: [0x6a72649c16d5246a207abdef78c8ce2148ed67c6c8a672bdac85e4c6ea2bdac8](https://polygonscan.com/tx/0x6a72649c16d5246a207abdef78c8ce2148ed67c6c8a672bdac85e4c6ea2bdac8)

Reviewer: ```suggestion
- `createAgent` function call example: [0x6a72649c16d5246a207abdef78c8ce2148ed67c6c8a672bdac85e4c6ea2bdac8](https://polygonscan.com/tx/0x6a72649c16d5246a207abdef78c8ce2148ed67c6c8a672bdac85e4c6ea2bdac8)
```
Similar suggestion to the other, and applicable in the other transactions as well.

- Update Agent function call example: 0x878cf23f0c3533941bc32f519a982fa294057a33ce52e8943bf03fe62ededeac

> - Disable Agent function call example: 0xe028ae4fdc771946b2fbef8c989f9074f317aa069d3a9b1d53086905bac7f8b8

Reviewer: ```suggestion
- Create Agent function call example: [0x6a72649c16d5246a207abdef78c8ce2148ed67c6c8a672bdac85e4c6ea2bdac8](https://polygonscan.com/tx/0x6a72649c16d5246a207abdef78c8ce2148ed67c6c8a672bdac85e4c6ea2bdac8)
- Update Agent function call example: [0x878cf23f0c3533941bc32f519a982fa294057a33ce52e8943bf03fe62ededeac](https://polygonscan.com/tx/0x878cf23f0c3533941bc32f519a982fa294057a33ce52e8943bf03fe62ededeac)
- Disable Agent function call example: [0xe028ae4fdc771946b2fbef8c989f9074f317aa069d3a9b1d53086905bac7f8b8](https://polygonscan.com/tx/0xe028ae4fdc771946b2fbef8c989f9074f317aa069d3a9b1d53086905bac7f8b8)
```
Team practice is to embed direct links to transactions in the relevant block explorer.
